languages/go/internal/host/reflect_test.go

```diff
@@ -42,9 +42,11 @@ func TestIsInstance(t *testing.T) {
 		t.Error("roles should be instances of the role class")
```

Comment:
Based on this error message, should L40 be `roleClass := reflect.TypeOf(Role)`?

Reply:
The error message is misleading but I believe the behavior here is correct when read with the accompanying inverse check on L48. I think `role constants should be instances of the Role class` might be a clearer statement of the test intent. 

We want to test that a constant instance of the newtype will match only the newtype `Role # reflect.TypeOf(Admin)` and not the wrapped type `String`. Prior to the change both of these `isInstance` calls would have returned true, but this had the side-effect of allowing for runtime confusion between unrelated types whose schemas were idnetical.

Reply:
So you're saying that `reflect.TypeOf(Admin)` returns the newtype `Role` whereas `reflect.TypeOf(Role)` would return `string`?

Reply:
`reflect.TypeOf(Role)` would make sense, but you can't actually reference types that way in Go, you can only get the type of a concrete value. Same reason in many places we have `myClass := reflect.TypeOf(MyClass{})` <- the `MyClass{}` is creating an instance of MyClass, and then we get the type of that new instance.

Reply:
> `reflect.TypeOf(Role)` would make sense, but you can't actually reference types that way in Go, you can only get the type of a concrete value. Same reason in many places we have `myClass := reflect.TypeOf(MyClass{})` <- the `MyClass{}` is creating an instance of MyClass, and then we get the type of that new instance.

Okay and at the risk of being obtuse, we can't do `reflect.TypeOf(Role{})` because `Role` is just a new type alias to `string` and `string` is a primitive instead of an interface? So instead we do `reflect.TypeOf(Admin)` because `Admin` points to a concrete (const) instance of `Role`?

Reply:
yep `reflect.TypeOf(Role{})` will barf because as a`NewType` of `string` it has no interface to satisfy this, so instead we use a constant instance which can be reflected.